pyfar/dsp/dsp.py

```diff
@@ -718,9 +718,9 @@ def _time_window_symmetric_interval_four(interval, window):
 
 
 def regularized_spectrum_inversion(
-        signal, freq_range,
+        signal, frequency_range=None,
```

Comment:
IHm, its a bit annoying that we have to change the default behaviour to make the change. It would be better if we could keep
```suggestion
        signal, frequency_range,
```
without the default `None`, but that would cause other problems. Any ideas anyone?

Reply:
What about using a decorator to intercept "freq_range" and rename the argument to "frequency_range" and throw a Deprecation warning? If that isn't too error-prone.

In that case we wouldn't have to touch the functions at all (apart from changing all freq_range to frequency_range).

Reply:
I think with fabians and my suggestions, it would already work as expected. No additional funtionallity and the old version would still work.

Reply:
Wouldnt' functions that require "frequency_range" as a positional argument throw a type error if "freq_range" is passed?

e.g.

```
pf.dsp.filter.erb_frequencies(freq_range=(20,20e3))

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], [line 1](vscode-notebook-cell:?execution_count=2&line=1)
----> [1](vscode-notebook-cell:?execution_count=2&line=1) pf.dsp.filter.erb_frequencies(freq_range=(20,20e3))

TypeError: erb_frequencies() missing 1 required positional argument: 'frequency_range' 
```

Reply:
Wouldn't that throw an error when "frequency_range" is a positional argument and only "freq_range" is passed?

e.g.
```
pf.dsp.regularized_spectrum_inversion(sweep, freq_range=(20, 20e3))

---------------------------------------------------------------------------
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[6], [line 1](vscode-notebook-cell:?execution_count=6&line=1)
----> [1](vscode-notebook-cell:?execution_count=6&line=1) pf.dsp.regularized_spectrum_inversion(sweep, freq_range=(20, 20e3))

TypeError: regularized_spectrum_inversion() missing 1 required positional argument: 'frequency_range'
```

with 

```def regularized_spectrum_inversion(
        signal, frequency_range,
        regu_outside=1., regu_inside=10**(-200/20), regu_final=None,
        normalized=True, *, freq_range=None):```

Reply:
Yes, the TypeError is the problem I thought about. Should we maybe discuss this in the weekly. The decorator sounds interesting to me, but I'm not sure if Anne has a better soloution in mind.

Reply:
ahh i see, thank you. this might help: https://stackoverflow.com/questions/74564140/best-practice-to-rename-a-method-parameter-in-a-deployed-python-module

Reply:
Yeah, that's where i got the idea with the decorator form. Should have mentioned it...

Reply:
In that case I think you could go straight ahead and implement it :)

Reply:
Ok, perfect!